libvmi/accessors.c

```diff
@@ -189,6 +202,20 @@ vmi_get_memsize(
     return vmi->allocated_ram_size;
 }
 
+void *
+vmi_get_memory_pfn(
```

Comment:
I don't think this is a good addition. LibVMI thus far deliberately avoided passing the raw pointer out so that it can clean up properly on shutdown. LibVMI also implements a cache that optimizes accesses to mapped-in pages so that if LibVMI functions need access to the page they don't have to remap it - which this new API now completely bypasses.

So this is a -1 from me. If the user needs this type of access to guest memory, the suggestion is to use libxc directly.

Reply:
We want to use libvmi with our introspection engine and we really actually do need this type of access to guest memory. What would be a good way to resolve the "map this page and write in it" case?
Our idea is to only use libvmi alone and without libxc in the same time.

Reply:
That's what the `vmi_write` is for: https://github.com/libvmi/libvmi/blob/master/libvmi/libvmi.h#L1514

Reply:
In our case it's more efficient to work directly with mem pages

Reply:
LibVMI by design doesn't give you direct access to the mem pages. You can either work with the existing API or you have to use libxc.

Reply:
Ok, I will drop this commit and re push